app/controllers/posts_controller.rb

```diff
@@ -3,7 +3,9 @@ class PostsController < ApplicationController
 
   def show
     @topic = Topic.find(params[:topic_id])
-    @post = Post.find(params[:id])
+    @post = @topic.posts.find(params[:id])
+    @comments = @post.comments
+    @comment = @post.comments.build
```

Comment:
Is the page showing one blank comment? If yes, consider modifying line 8:
```diff
-    @comment = @post.comments.build
+    @comment = Comment.new
```
The `@post.comments.build` call will add a blank, unsaved comment to the end of `@post.comments`.